lib/msf/core/auxiliary/report_summary.rb

```diff
@@ -0,0 +1,132 @@
+
+      def create_credential_login(credential_data)
+        credential = credential_data[:core].to_credential
+        @@report[rhost] = { successful_logins: [] }
+        @@report[rhost][:successful_logins] << credential
+        super
+      end
+
+      # I think this will work for our new modules, but I don't know if it'll work
+      # for other modules that might not use 'session_setup' convention;
+      # i.e. you might need to hijack start_session, or create_session, or some other
+      # lower level building block that you'll have access to
+      def start_session(obj, info, ds_merge, crlf = false, sock = nil, sess = nil)
+        result = super
+        @@report[rhost].merge!({ successful_sessions: [] })
+        @@report[rhost][:successful_sessions] << result
+        result
+      end
+
+      private
+
+      def print_report_summary
+        report = @@report
+
+        conditional_verbose_output(report.keys.count)
+
+        logins = report.flat_map { |_k, v| v[:successful_logins] }.compact
+        sessions = report.flat_map { |_k, v| v[:successful_sessions] }.compact
+
+        # TODO - This was previously within the run command that was initially added as a workaround
+        # return results unless framework.features.enabled?(Msf::FeatureManager::MSSQL_SESSION_TYPE)
+
+        print_status("Bruteforce completed, #{logins.size} #{logins.size == 1 ? 'credential was' : 'credentials were'} successful.")
+        if datastore['CreateSession']
+          print_status("#{sessions.size} MySQL #{sessions.size == 1 ? 'session was' : 'sessions were'} opened successfully.")
+        else
+          print_status('You can open an MySQL session with these credentials and %grnCreateSession%clr set to true')
+        end
+
+        show_successful_logins(report)
+        report
+      end
+
+      # Logic to detect if the ShowSuccessLogins datastore option has been set
+      #
+      # @param [Hash] report Host mapped to successful logins and sessions
+      # @return [String] Rex::Text::Table containing successful logins
+      def show_successful_logins(report)
+        if datastore['ShowSuccessfulLogins'] == true
+          successful_logins_to_table(report)
+        end
+      end
+
+      # The idea here is to add a hybrid approach for scanner modules
+      # If only one host is scanned a more verbose output is useful to the user
+      # If scanning multiple hosts we would want more lightweight information
+      #
+      # @param [Object] host_count The number of hosts
+      def conditional_verbose_output(host_count)
+        if host_count == 1
+          datastore['Verbose'] = true
+        end
+      end
+
+      # Takes the login/session results and converts them into a Rex::Text::Table format
+      #
+      # @param report [Hash{String => [Metasploit::Framework::LoginScanner::Result, Msf::Sessions]}]
+      # @return [String] Rex::Text::Table containing successful logins
+      def successful_logins_to_table(report)
```

Comment:
Can we change the hide the table if there's no successful logins

```diff
[*] 0 MySQL sessions were opened successfully.
-
-Successful logins
-=================
-
-    Host  Public  Private
-    ----  ------  -------
-
-
[*] Auxiliary module execution completed
```

Reply:
Good, call. I'll get that sorted.